### ISSUE TITLE
v3.9.0 fails on linting (v3.8.0 still works)

### ISSUE THREAD

**Issue by marcofranssen** [helm/chart-testing#580]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

BUG REPORT

**Version of Helm and Kubernetes**:

- ct v3.9.0
- helm v3.12.0

**What happened**:

https://github.com/spiffe/helm-charts/actions/runs/5602051333/jobs/10246787989?pr=324#step:6:20

**What you expected to happen**:

To have the pipeline succeed just like it does with the v3.8.0 version of `ct`.

**How to reproduce it** (as minimally and precisely as possible):

1. Use chart-testing 3.9.0
2. Use the github action checkout action using `fetch: 0`
3. run `ct lint --config ct.yaml --target-branch $(TARGET_BRANCH) --check-version-increment=false`
4. now run the same with 3.8.0 and you will see it still works as expected.

**Anything else we need to know**:

See here the working variant with v3.8.0

**Comment by alacuku** [comment]
The issue seems to be related to: https://github.com/helm/chart-testing/pull/552. Since it is passing down the `extra-helm-args` to `helm lint` it could happen that flags between differen helm commands are incompatible such as the `--timeout` flag which is a legit one for `helm install` but makes `helm lint` to fail.

**Comment by joejulian** [comment]
Perhaps we should have more than one helm-extra-args, ie: helm-install-extra-args, helm-upgrade-extra-args, helm-lint-extra-args, etc. Same as above, ct 3.9.0 now breaks because `helm lint --timeout 900s` is invalid.

**Comment by joejulian** [comment]
Which was also suggested in #540

**Comment by AndersBennedsgaard** [comment]
@joejulian I agree, that sounds like a good solution going forward, but we should probably keep the `--helm-extra-args` such that you are able to reduce duplicate settings.

P.s. I haven't actually had issues with 3.9.0 yet, so you only just made me aware of this (sorry for the inconveniences my change has brought 😞 )

**Comment by jouve** [comment]
this looks like a duplicate of #577 , so I suggest the same [workaround](https://github.com/helm/chart-testing/issues/577#issuecomment-1645205549) for gitlab-ci :
before_script:
  - git branch "$target_branch" "origin/$target_branch"

**Comment by prashant-shahi** [comment]
> ct 3.9.0 now breaks because helm lint --timeout 900s is invalid

@joejulian @AndersBennedsgaard I am facing same issue. Fails in `3.9.0` and `3.10.0`. How did you guys overcome this?

**Comment by cpanato** [comment]
there is a fix for that https://github.com/helm/chart-testing/pull/605
